### ISSUE TITLE
What's the difference between these log probs? 

### ISSUE THREAD

**Issue by Hap-Zhang** [NVIDIA/TensorRT-LLM#1132]
Hi,all
I see tensorrtllm can output these probabilities: "cum_log_probs", "output_log_probs", "context_logits", "generation_logits". What is the correlation between these probabilities?And where can I find a detailed explanation?
```[tasklist]
### Tasks
```

**Comment by byshiue** [comment]
`context_logits` and `generation_logits` are logits of context tokens and generation tokens. 

`output_log_probs` is log of probability (probabilities are computed from logits) of generated token.

`cum_log_probs` is cumulative log probability of generated tokens.

**Comment by Hap-Zhang** [comment]
@byshiue Thank you very much for your reply. However, when i set **"return_log_probs"** to true, only **"cum_log_probs"** can returned normally and **"output_log_probs"** is all zero. Should the cumulative multiplication of **"output_log_probs"** be equal to **"cum_log_probs"**?

**Comment by nekorobov** [comment]
Hi @Hap-Zhang,

do you use TopK=1 and set `normalizeLogProbs=true` (I.e default value) to the params to the `GptManager`? If so, then set [this flag](https://github.com/NVIDIA/TensorRT-LLM/blob/3c373ebc5b5caf7e41198125131a153f3df08f09/cpp/include/tensorrt_llm/batch_manager/trtGptModelOptionalParams.h#L45) to false to have cumulative multiplication of `output_log_probs` being equal to `cum_log_probs`.
See this [comment](https://github.com/NVIDIA/TensorRT-LLM/blob/3c373ebc5b5caf7e41198125131a153f3df08f09/cpp/tensorrt_llm/kernels/samplingTopKKernels.cu#L229) for log probs normalization explanation.

**Comment by Hap-Zhang** [comment]
Hi， @nekorobov ,
Thank you very much! I use tensorrt-llm backend in triton server, and will have a try by set "normalize_log_probs" to false.
